scalableminds/webknossos
# Improve upload UI

This PR reworks the Upload UI and additionally adds the check, whether a dataset needs to be converted to the front end. Therefore the user only has to set the dataset scale if necessary.

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test (local):
- Open the dataset upload ui via the dashboard.
- Upload a new dataset in zip format.
- Only if multiple datastores are available, the user can select a datastore to which the dataset should be uploaded.
- Additionally, the user should only be required to enter the dataset scale, if the format of the dataset is not wkw.
- Upon starting the upload a modal should open up, displaying the uploading process or errors.
- After the upload finished, the content of the modal should change and suggest the user to either view the dataset, go back to the dashboard or go to the settings of the uploaded dataset. Each of those links should work as expected.
- Additionally, after the upload finished successfully the modal should now be closable by using the ok button that was previously disabled.

- Another to test is whether the checking of the zip file content for a wkw file takes too much memory. In my test cases, this never was a problem.

### Issues:
- fixes #5018, #5001

------
